backend/tests/integration-async/disambiguation/test_disambiguation_tasks.py

```diff
@@ -1253,3 +1255,50 @@ def assert_disambiguation_task():
         )
 
     retry_until_pass(assert_disambiguation_task, retry_interval=5)
+
+
+def test_editor_lock_is_created_when_disambiguation_runs(
+    inspire_app, clean_celery_session, enable_disambiguation
+):
+    redis_url = current_app.config.get("CACHE_REDIS_URL")
+    redis = StrictRedis.from_url(redis_url)
+    author_data = faker.record("aut", with_control_number=True)
+    author_data.update(
+        {
+            "name": {"value": "Brian Gross"},
+            "ids": [{"schema": "INSPIRE BAI", "value": "J.M.Maldacena.1"}],
+            "email_addresses": [{"current": True, "value": "test@test.com"}],
+        }
+    )
+    author_record = InspireRecord.create(author_data)
+    db.session.commit()
+
+    def assert_authors_records_exist_in_es():
+        author_record_from_es = InspireSearch.get_record_data_from_es(author_record)
+        assert author_record_from_es
+
+    retry_until_pass(assert_authors_records_exist_in_es)
+
+    literature_data = faker.record("lit", with_control_number=True)
+    literature_data.update(
+        {
+            "authors": [
+                {
+                    "full_name": "Brian Gross",
+                    "ids": [{"schema": "INSPIRE BAI", "value": "J.M.Maldacena.1"}],
+                    "emails": ["test@test.com"],
+                }
+            ]
+        }
+    )
+    literature_record = LiteratureRecord.create(literature_data)
+    db.session.commit()
+    version_id = literature_record.model.version_id
+
+    def assert_lock_in_redis():
+        expected_hash_name = (
+            f"editor-task-lock:{literature_record['control_number']}@{version_id}"
+        )
+        assert redis.hgetall(expected_hash_name)
+
+    retry_until_pass(assert_lock_in_redis, retry_interval=1)
```

Comment:
won't this be flaky? can you be sure that the disambiguation task doesn't finish before you've checked the condition here (in which case the lock has already been cleared)?

Reply:
I don't have any other idea how to test it. For me this test never failed, but I can remove it if you want.